cub/cub/device/device_reduce.cuh

```diff
@@ -304,6 +307,8 @@ struct DeviceReduce
       NumItemsT num_items,
       cudaStream_t stream = 0)
   {
+    CUB_NVTX_RANGE_SCOPE_IF(d_temp_storage);
```

Comment:
concern: I just double checked and just using `__func__` with a member function like this will mean this range will show up named `Sum`. People won't really be able to tell that this is a `DeviceReduce::Sum`. 

https://godbolt.org/z/bf5jf1nfj

I don't have a good suggestion on how to conveniently include a more meaningful name for the range :/

Reply:
Right. @gevtushenko already told me, he would like to have the fully qualified name `cub::DeviceReduce::Sum`. I have not found a way yet to get this string automatically (and also portably), so we may have to spell it out manually.

Reply:
One idea to reduce the amount of typing would be to add a member/function to each algorithm type that just returns a string of its name.  

Something like this?
```
struct DeviceReduce{
   auto name(){ return "DeviceReduce"; };

   static auto Sum(...){
      CUB_NVTX_RANGE(name());
};
```

Reply:
That would work, but then, e.g., `SortPairs`, `SortPairsDecending`, `SortKeys`, `SortKeysDecending` would all be reported as `DeviceRadixSort`, which is technically true, but less helpful IMO. I would rather incur more effort on my side to give a better user experience.

Reply:
I discussed the issue with @gevtushenko and we agreed on using precise names except for sorting functions.